pyne/alara.py

```diff
@@ -890,5 +892,242 @@ def _find_phsrc_dc(idc, phtn_src_dc):
                 return phtn_src_dc[i]
         raise ValueError('Decay time {0} not found in phtn_src file'.format(idc)) 
 
+def _gt_normalize(neutron_spectrum):
+    """
+    Function that normalizes the neutron spectrum for ALARA fluxin file
+    """
+    tol = 1E-8
+    total = float(np.sum(neutron_spectrum))
+    if abs(total - 1.0) > tol:
+        warn("Normalizing neutron spectrum")
+        neutron_spectrum = [x / total for x in neutron_spectrum]
+    return neutron_spectrum
+
+def _gt_write_matlib(mats, filename):
+    """
+    Function that writes matlib file for ALARA run
+    """
+    s = ""
+    for m, mat in enumerate(mats):
+        s += mat.alara()
+        s += "\n"
+    with open(filename, 'w') as f:
+        f.write(s)
 
+def _gt_write_fluxin(fluxes, fluxin_file):
+    """
+    Function that writes ALARA fluxin file
+    """
+    s = ""
+    for flux in fluxes:
+        for i, fl in enumerate(reversed(flux)):
+            s += "{0:.6E} ".format(fl)
+            if (i + 1) % 6 == 0:
+                s += '\n'
+        s += '\n\n'
+    with open(fluxin_file, 'w') as f:
+        f.write(s)
+
+def _gt_write_inp(run_dir, data_dir, mats, num_n_groups, flux_magnitudes,
+                  irr_times, decay_times, input_file, matlib_file,
+                  fluxin_file, phtn_src_file, num_p_groups):
+    """
+    Function that writes ALARA input file
+    """
+    num_zones = len(mats) * (num_n_groups)
+    s = "geometry rectangular\n\nvolume\n"
```

Comment:
This  might be more readable with a long multiline string and a `template` or `format`